Nvshmem add new versions

bvanessen opened the pull request.

bvanessen committed:
Added new versions of NVSHMEM.  Enabled CMake support.  Added variant for IBRC.

bvanessen committed:
Fixed how the latest nvshmem tarball is found.

tldahlgren reviewed (commented):
I can confirm the sha256 for `2.8.0-3` but am getting `Could not find any remote versions for nvshmem` for `2.9.0-2`.

bvanessen committed:
Testing new compile and runtime environment variables.

bvanessen commented:
@jdinan
@Seth5151

Here is the current spack PR.

Seth5141 commented on var/spack/repos/builtin/packages/nvshmem/package.py:
```diff
@@ -73,12 +86,44 @@ def setup_build_environment(self, env):
+        if "+ibrc" in self.spec:
+            env.set("NVSHMEM_IBRC_SUPPORT", "1")
 
+    @when("@2.6:")
```
Is the @2.9 inside of setup_run_environment extraneous? I thought the @when("@2.6:") would only run this function for 2.6? (Just getting a grasp of spack syntax).

Seth5141 commented on var/spack/repos/builtin/packages/nvshmem/package.py:
```diff
@@ -38,12 +40,23 @@ class Nvshmem(MakefilePackage, CudaPackage):
         directory = ver_str.split("-")[0]
-        url_fmt = "https://developer.download.nvidia.com/compute/redist/nvshmem/{0}/source/nvshmem_src_{1}.txz"
+        if self.spec.satisfies("@2.9:"):
+            url_fmt = "https://developer.download.nvidia.com/compute/redist/nvshmem/{0}/source/nvshmem_src_{1}.tar.xz"
```
Sorry about this regression. I can work internally to fix this filename for 2.9 and going forward.

Seth5141 commented on var/spack/repos/builtin/packages/nvshmem/package.py:
```diff
@@ -38,12 +40,23 @@ class Nvshmem(MakefilePackage, CudaPackage):
+    )
     conflicts("~cuda")
 
+    build_system("makefile", "cmake", default="makefile")
```
FYI going forward from 2.9 we are making cmake the default build system for NVSHMEM (Makefile still supported in 2.9).

bvanessen committed:
Added support for building NVSHMEM with libfabric support.  Added
environment variables for versions 2.6:.  Setup both runtime and
dependent runtime environment variables.

bvanessen committed:
Updated flags for including NVSHMEM.

kylosus commented:
Hello! Any plans to merge this anytime soon? We have NVSHMEM 2.10 now

alalazo commented:
@bvanessen Are you still interested in this PR?

alalazo commented:
Closing, since there has been no activity in a while. Feel free to reopen at / migrate to https://github.com/spack/spack-packages if you want to continue work on this.
